### PR TITLE
version 1

### PR DESCRIPTION
**Pull request by artbycrunk**
version 1 of custom keycap inspired by iris defaults

### CODE REVIEW

File: keyboards/iris/keymaps/saviof/keymap.c [hunk 1]
```diff
@@ -0,0 +1,115 @@
+#include "iris.h"
```
Comment (noroadsleft) [inline keyboards/iris/keymaps/saviof/keymap.c]: Replace Lines 1-3 with `#include QMK_KEYBOARD_H`.

File: keyboards/iris/keymaps/saviof/keymap.c [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  TD_SPC_ENT = 0
+};
+
+//Tap Dance Definitions
+qk_tap_dance_action_t tap_dance_actions[] = {
+    [TD_SPC_ENT] = ACTION_TAP_DANCE_DOUBLE(KC_SPC, KC_ENT)
+};
+
+#define KC_ KC_TRNS
+#define _______ KC_TRNS
```
Comment (noroadsleft) [inline keyboards/iris/keymaps/saviof/keymap.c]: This line isn't necessary and can be safely deleted; QMK defines this automatically.

https://github.com/qmk/qmk_firmware/blob/35e76539e7257a1932ec76882892b999e21c9d4f/quantum/quantum_keycodes.h#L29-L31

File: keyboards/iris/keymaps/saviof/rules.mk [hunk 3]
```diff
@@ -0,0 +1,7 @@
+RGBLIGHT_ENABLE = yes
+BACKLIGHT_ENABLE = yes
+TAP_DANCE_ENABLE=yes
+
+ifndef QUANTUM_DIR
```
Comment (noroadsleft) [inline keyboards/iris/keymaps/saviof/rules.mk]: Lines 5-7 should be deleted. This is leftover code from an old `make` system.

**Comment by artbycrunk** [comment]
updated based on feedback

File: keyboards/iris/keymaps/saviof/keymap.c [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+enum {
+  TD_SPC_ENT = 0
+};
+
+//Tap Dance Definitions
+qk_tap_dance_action_t tap_dance_actions[] = {
+    [TD_SPC_ENT] = ACTION_TAP_DANCE_DOUBLE(KC_SPC, KC_ENT)
+};
+
+#define KC_LOWR LT(_LOWER, KC_LEFT) //LOWER
```
Comment (noroadsleft) [inline keyboards/iris/keymaps/saviof/keymap.c]: You need to add `#define KC_ KC_TRNS` above Line 27.

It was only the `#define _______ KC_TRNS` that needed deletion before.

File: keyboards/iris/keymaps/saviof/config.h [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/iris/keymaps/saviof/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/iris/keymaps/saviof/config.h [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+#define CONFIG_USER_H
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/iris/keymaps/saviof/config.h]: This line is not actually needed, and should be removed. 

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/iris/keymaps/saviof/keymap.c [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  //|----+----+----+----+----+----+----.    ,----|----+----+----+----+----+----|
+     BL_S,RST ,    ,    ,    ,    ,    ,         ,    ,    ,    ,    ,    ,    ,
+  //`----+----+----+--+-+----+----+----/    \----+----+----+----+----+----+----'
+                           ,    ,    ,             ,    ,
+  //                  `----+----+----'        `----+----+----'
+  )
+
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/iris/keymaps/saviof/keymap.c]: Could you remove this function, especially since you're not using it?

**Comment by artbycrunk** [comment]
updated based on feedback

**Comment by artbycrunk** [comment]
updated based on feedback

**Comment by artbycrunk** [comment]
updated based on feedback

**Comment by drashna** [comment]
Thanks! Looks good!

**Pull request merged**